app/controllers/DeclarationDetailController.scala

```diff
@@ -41,41 +43,109 @@ class DeclarationDetailController @Inject()(authenticate: IdentifierAction,
                                             view: cash_account_declaration_details,
                                             cashAccountUtils: CashAccountUtils,
-                                            noTransactions: cash_transactions_no_result
+                                            noTransactionsView: cash_transactions_no_result
                                            )(implicit executionContext: ExecutionContext,
-                                             val appConfig: AppConfig
+                                             appConfig: AppConfig
                                            ) extends FrontendController(mcc) with I18nSupport with Logging {
 
-  def displayDetails(ref: String, page: Option[Int]): Action[AnyContent] =
-    (authenticate andThen verifyEmail).async { implicit request =>
-
-      apiConnector.getCashAccount(request.eori).flatMap {
+  def displaySearchDetails(ref: String, page: Option[Int], searchInput: String): Action[AnyContent] =
+    displayDetails(ref, page, cameViaSearch = true, searchInput)
 
+  def displayDetails(ref: String,
+                     page: Option[Int],
+                     cameViaSearch: Boolean = false,
+                     searchInput: String = emptyString
+                    ): Action[AnyContent] = (authenticate andThen verifyEmail).async { implicit request =>
+    for {
+      accountOpt <- apiConnector.getCashAccount(request.eori)
+      result <- accountOpt match {
         case Some(account) =>
           val (from, to) = cashAccountUtils.transactionDateRange()
-          retrieveCashAccountTransactionAndDisplay(account, from, to, ref, page)
-
+          retrieveCashAccountTransactionAndDisplay(account, from, to, ref, page, cameViaSearch, searchInput)
         case None => Future.successful(NotFound(errorHandler.notFoundTemplate))
       }
+    } yield result
+  }
+
+  def handleSearchRequest(page: Option[Int], searchInput: String)
+                         (implicit request: IdentifierRequest[AnyContent]): Future[Result] = {
+    if (isValidMRNUCR(searchInput) || isValidPayment(searchInput)) {
+      val searchType = determineSearchType(searchInput)
+      searchDeclarations(page, searchInput, searchType)
+    } else {
+      Future.successful(NotFound(errorHandler.notFoundTemplate))
     }
+  }
+
+  def determineSearchType(searchInput: String): Declaration => Boolean = {
+    if (isValidMRNUCR(searchInput)) {
+      declaration =>
+        declaration.movementReferenceNumber.contains(searchInput) ||
+          declaration.declarantReference.contains(searchInput)
+    } else {
+      val paymentAmount = parsePaymentAmount(searchInput)
+      (declaration: Declaration) => declaration.amount == paymentAmount
+    }
+  }
+
+  def isValidMRNUCR(value: String): Boolean =
+    mrnUCRRegex.findFirstIn(value).isDefined || superMRNUCRRegex.findFirstIn(value).isDefined
+
+  def isValidPayment(value: String): Boolean =
```

Comment:
I don't think this needs to be two separate lines, although I appreciate the consistency as it matches the other defs. Not sure if we just want to move it two one or not?